File: landing-page/examples/java/Social.java
// javac -cp neo4j-java-driver*.jar:. Network.java
// java -cp neo4j-java-driver*.jar:. Network

import org.neo4j.driver.v1.*;
import static org.neo4j.driver.v1.Values.parameters;

import java.util.List;
import static java.util.Arrays.asList;
import static java.util.Collections.singletonMap;

public class Social {

    public static void main(String...args) {
        
        Config noSSL = Config.build().withEncryptionLevel(Config.EncryptionLevel.NONE).toConfig();
        Driver driver = GraphDatabase.driver("bolt://localhost",AuthTokens.basic("neo4j","test"),noSSL); // <password>
        try (Session session = driver.session()) {

              List data = 
              asList(asList("Jim","Mike"),asList("Jim","Billy"),asList("Anna","Jim"),
              asList("Anna","Mike"),asList("Sally","Anna"),asList("Joe","Sally"),
              asList("Joe","Bob"),asList("Bob","Sally"));
    
              String insertQuery = "UNWIND {pairs} as pair " +
               "MERGE (p1:Person {name:pair[0]}) " +
               "MERGE (p2:Person {name:pair[1]}) " +
               "MERGE (p1)-[:KNOWS]-(p2);";
      
               session.run(insertQuery,singletonMap("pairs",data)).consume();
          
               StatementResult result;
            
               String foafQuery = 
               " MATCH (person:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf) "+
               " WHERE person.name = {name} " +
               "   AND NOT (person)-[:KNOWS]-(foaf) " +
               " RETURN foaf.name AS name ";
               result = session.run(foafQuery, parameters("name","Joe"));
               while (result.hasNext()) System.out.println(result.next().get("name"));
      
               String commonFriendsQuery =
               "MATCH (user:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf:Person) " +
               " WHERE user.name = {from} AND foaf.name = {to} " +
               " RETURN friend.name AS friend";
               result = session.run(commonFriendsQuery, parameters("from","Joe","to","Sally"));
               while (result.hasNext()) System.out.println(result.next().get("friend"));
      
               String connectingPathsQuery =
               "MATCH path = shortestPath((p1:Person)-[:KNOWS*..6]-(p2:Person)) " +
               " WHERE p1.name = {from} AND p2.name = {to} " +
               " RETURN [n IN nodes(path) | n.name] as names";
               result = session.run(connectingPathsQuery, parameters("from","Joe","to","Billy"));
               while (result.hasNext()) System.out.println(result.next().get("names"));

        }
    }
}

File: landing-page/examples/dotnet/ConsoleApplication1/examples/Social.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Neo4j.Driver.V1;

class SocialProgram
{
    private static void SocialMain(string[] args)
    {
        using (var s = new Social())
        {
            s.Setup();

            var friendsOfFriends = s.FriendsOfAFriend("Joe");
            Console.WriteLine($"Joe's friends (of friends)");
            foreach (var fof in friendsOfFriends)
            {
                Console.WriteLine($"\t{fof["foaf"].As<INode>()["name"].As<string>()}");
            }

            Console.WriteLine();

            var commonFriends = s.CommonFriends("Joe", "Sally");
            Console.WriteLine("Joe and Sally's common friends");
            foreach (var friend in commonFriends)
            {
                Console.WriteLine($"\t{friend["friend"].As<string>()}");
            }

            Console.WriteLine();

            var connectingNames = s.ConnectingPaths("Joe", "Billy");
            Console.WriteLine("Path to Billy");
            foreach (var record in connectingNames)
            {
                var path = record["path"].As<IPath>();
                foreach (var friend in path.Nodes)
                {
                    Console.WriteLine($"\t{friend["name"].As<string>()}");
                }
            }
        }
    }
}

public class Social :IDisposable
{
    private readonly IDriver _driver;

    public Social()
    {
        _driver = GraphDatabase.Driver(new Uri("bolt://localhost:7687"), AuthTokens.Basic("neo4j", "neo4j"));
    }

    public void Setup()
    {
        var peopleList = new List<string[]>
        {
            new[] {"Jim", "Mike"}, new[] {"Jim", "Billy"}, new[] {"Anna", "Jim"},
            new[] {"Anna", "Mike"}, new[] {"Sally", "Anna"}, new[] {"Joe", "Sally"},
            new[] {"Joe", "Bob"}, new[] {"Bob", "Sally"}
        };

        using (var session = _driver.Session())
        {
            var result = session.Run(
                "UNWIND {people} AS pair " +
                "MERGE (u1:Person { name: pair[0] }) " +
                "MERGE (u2:Person { name: pair[1] }) " +
                "MERGE (u1)-[:KNOWS]->(u2)", new {people = peopleList});
            result.ToList();
        }
    }

    public List<IRecord> FriendsOfAFriend(string personName)
    {
        using (var session = _driver.Session())
        {
            var result = session.Run(
                "MATCH (p:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf)" +
                "WHERE p.name = {p1}" +
                "AND NOT (p) -[:KNOWS]-(foaf)" +
                "RETURN foaf", 
                new {p1 = personName});

            return result.ToList();
        }
    }

    public List<IRecord> CommonFriends(string personName1, string personName2)
    {
        using (var session = _driver.Session())
        {
            var result = session.Run(
                "MATCH (p:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf:Person)" +
                "WHERE p.name = {p1}" +
                "AND foaf.name = {p2}" +
                "RETURN friend.name AS friend",
                new { p1 = personName1, p2 = personName2 });

            return result.ToList();
        }
    }

    public List<IRecord> ConnectingPaths(string personName1, string personName2)
    {
        using (var session = _driver.Session())
        {
            var result = session.Run(
                "MATCH path = shortestPath((p1:Person)-[:KNOWS*..6]-(p2:Person))" +
                "WHERE p1.name = {p1}" +
                "AND p2.name = {p2}" +
                "RETURN path",
                new { p1 = personName1, p2 = personName2 });

            return result.ToList();
        }
    }

    public void Dispose()
    {
        _driver?.Dispose();
    }
}